File: dotnet/Razorvine.Pyrolite/Pyrolite/Pickle/IObjectConstructor.cs
/* part of Pyrolite, by Irmen de Jong (irmen@razorvine.net) */

// ReSharper disable InconsistentNaming
namespace Razorvine.Pickle
{

/// <summary>
/// Interface for object Constructors that are used by the unpickler
/// to create instances of non-primitive or custom classes.
/// </summary>
public interface IObjectConstructor {

	/**
	 * Create an object. Use the given args as parameters for the constructor.
	 */
	object construct(object[] args);
}

}

File: java/src/main/java/net/razorvine/pickle/IObjectConstructor.java
package net.razorvine.pickle;

/**
 * Interface for Object Constructors that are used by the unpickler
 * to create instances of non-primitive or custom classes.
 *
 * @author Irmen de Jong (irmen@razorvine.net)
 */
public interface IObjectConstructor {

	/**
	 * Create an object. Use the given args as parameters for the constructor.
	 */
	Object construct(Object[] args) throws PickleException;
}
